src/js/protocols/webusbdfu.js

```diff
@@ -141,27 +141,20 @@ class WEBUSBDFU_protocol extends EventTarget {
         // reset progress bar to initial state
         TABS.firmware_flasher.flashingMessage(null, TABS.firmware_flasher.FLASH_MESSAGE_TYPES.NEUTRAL).flashProgress(0);
 
-        navigator.usb
-        .requestDevice(usbDevices)
-        .then(selectedDevice => {
-            console.log(`Product name: ${selectedDevice.productName}`);
-            console.log(`USB DFU detected with ID: ${selectedDevice}`);
-            this.usbDevice = selectedDevice;
-            console.log(`WebUSB Version: ${this.usbDevice.deviceVersionMajor}.${this.usbDevice.deviceVersionMinor}.${this.usbDevice.deviceVersionSubminor}`);
-            return this.openDevice();
-        })
-        .catch((e) => {
-            console.error(`There is no device. ${e}`);
-            gui_log(i18n.getMessage('stm32UsbDfuNotFound'));
-        });
+        const devices = await this.getDevices();
+        const deviceFound = devices.find(device => device.path === devicePath);
+        this.usbDevice = deviceFound.port;
```

Comment:
Suspect we need to use the object `deviceFound` itself here.

Reply:
The deviceFound.port contains the full device object itself. We create a wrapper object with several fields and this is the full object.